### DISCUSSION TITLE
Basic usage question

### DISCUSSION THREAD

**Discussion by Aenge** [ClearFoundry/ClearScript#242] [Q&A] [answered]
Hello, I'm trying to get up and running with ClearScript. I'm having trouble accessing class members in the script.

![image](https://user-images.githubusercontent.com/38081041/111242520-bb30d500-85d5-11eb-92d1-a1b6b3bb39d4.png)

**Comment by ClearScriptLib** [comment, marked as answer]
Hello @Aenge,

We ran the code above and it printed "0" as expected. Can you double-check your results?

If the problem persists, please tell us about your setup – OS, architecture, ClearScript version, .NET runtime, etc.

Thanks!

**Reply by Aenge** [reply]
Windows 10 x64
ClearScript v7.1.1
Tried .NET 5.0 and .NET Core 3.1, same results with both

Edit: The issue seems to be when the player class is owned by the Program class. If I move it outside of Program, it works as expected

**Reply by ClearScriptLib** [reply]
Ah, OK. Your edit provided the clue we needed.

ClearScript strictly enforces member visibility, and as coded above, there's no public path to `player` (since `Program` is internal). So even though you're handing over a `player` instance, script code has no access to `player`'s defined members, and can only see the members of its most derived fully public base class, which in this case is just `object`.

There are several ways to fix this, the simplest being to make `Program` public. You could also use [`AccessContext`](https://clearscript.clearfoundry.net/Reference/html/P_Microsoft_ClearScript_ScriptEngine_AccessContext.htm) to declare that  your script code runs within `Program`'s visibility scope, but that's seldom the best option.

Good luck!

**Reply by Aenge** [reply]
Such a simple thing - thanks for the clarification.
